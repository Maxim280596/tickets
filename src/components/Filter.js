import React from 'react';
import {Button, Col, Row} from 'react-bootstrap';
import styled from 'styled-components';


const StyledButton = styled.button `
  width:50%;
  height:25%;
  margin-bottom: 20px;
  border-radius:10px;
  
`;


export  function Filter () {


  return(
    < >
    
    <StyledButton variant="primary" size="lg">Самый дешевый</StyledButton>
    <StyledButton variant="primary" size="lg">Самый быстрый</StyledButton>
    
    </>
  )
}