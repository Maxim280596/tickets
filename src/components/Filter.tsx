import React, { useState } from 'react';
import { useDispatch } from 'react-redux';

import { StyledButtonFast, StyledButtonPrice } from '../styled/StyledFilter';
import {
  filteringFast,
  filteringPrice,
} from '../redux/ducks/ticketsReduser';


const Filter = () => {
  const dispatch = useDispatch();
  const [price, setPrice] = useState(false);
  const [fast, setFast] = useState(false);
  const filterPrice = () => {
    dispatch(filteringPrice());
    setPrice(true);
    setFast(false);
  };

  const filterFast = () => {
    dispatch(filteringFast());
    setPrice(false);
    setFast(true);
  };

  return (
    <>
      <StyledButtonPrice onClick={filterPrice} active={price}>
        САМЫЙ ДЕШЕВЫЙ
      </StyledButtonPrice>
      <StyledButtonFast onClick={filterFast} active={fast}>
        САМЫЙ БЫСТРЫЙ
      </StyledButtonFast>
    </>
  );
};

export default Filter;
