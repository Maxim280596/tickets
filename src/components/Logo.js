import React from 'react';
import { Row, Container } from 'react-bootstrap';

import Logotype from '../logo.png';
import { StyledLogo, StyledLogoCol } from '../styled/StyledLogo';

const Logo = () => (
  <Container>
    <Row>
      <StyledLogoCol>
        <StyledLogo src={Logotype} alt="Logo" />
      </StyledLogoCol>
    </Row>
  </Container>
);

export default Logo;
