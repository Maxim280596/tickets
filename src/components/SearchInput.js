import React from 'react';

import { StyledCheckbox } from '../styled/StyledInput';

export const SearchInput = ({
  inline,
  label,
  type,
  id,
  onChange,
  checked,
  key,
  name,
}) => {
  return (
    <div>
      <StyledCheckbox
        name={name}
        key={key}
        onChange={onChange}
        id={id}
        inline={inline}
        label={label}
        type={type}
        checked={checked}
      />
    </div>
  );
};
