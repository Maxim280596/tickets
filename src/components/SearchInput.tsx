import React from 'react';
import { Field } from 'formik';

import { StyledCheckbox } from '../styled/StyledInput';

export const SearchInput: React.FC<{
  toggleCheckbox:any;
  name: String;
  options: { id: number; path: string; value: string; checked: boolean; length: number; }[];
}> = ({ toggleCheckbox, name, options }) => {
  return (
    <Field
      name={name}
      render={(array: []) => (
        <div>
          {options.map((item:{id: number; path: string; value: string; checked: boolean; length: number;}) => (
            <StyledCheckbox
              key={item.id}
              onChange={() => toggleCheckbox(item)}
              type="checkbox"
              value={item.value}
              checked={item.checked}
              label={item.value}
              length={item.length}
            />
          ))}
        </div>
      )}
    />
  );
};
