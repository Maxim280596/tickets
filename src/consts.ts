export const checkboxOptions = [
  { id: 1, path: '/', value: 'Все', checked: true, length: 10 },
  {
    id: 2,
    path: '/no_stops',
    value: 'Без пересадки',
    checked: false,
    length: 0,
  },
  { id: 3, path: '/one_stop', value: '1 пересадка', checked: false, length: 1 },
  {
    id: 4,
    path: '/two_stops',
    value: '2 пересадки',
    checked: false,
    length: 2,
  },
  {
    id: 5,
    path: '/three_stops',
    value: '3 пересадки',
    checked: false,
    length: 3,
  },
];
