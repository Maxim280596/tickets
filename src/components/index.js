import { Search } from "./Search/Search";
import { Filter } from "./Filter/Filter";
import { TicketsList } from "./TicketList/TicketsList";
import { Logo } from "./Logo/Logo";

export {Logo,TicketsList,Filter,Search};