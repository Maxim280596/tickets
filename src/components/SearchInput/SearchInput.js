import React from 'react';
import {StyledCheckbox} from '../../styled/SearchInput/StyledInput'


export const SearchInput = ({inline,label,type, id , onClick}) => {
  
    
  return (
    
        <div>
          <StyledCheckbox id={id} inline={inline} label={label} type={type} />
        </div>
        
  )
}