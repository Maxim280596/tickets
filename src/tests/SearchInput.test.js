import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import {SearchInput} from '../components/SearchInput';
import {checkboxOptions} from '../consts.ts'

describe('Testing <SearchInput/>', () => {
  it('SearchInput have rendered correctly', ()=>{
    const searchinput = shallow(<SearchInput options={checkboxOptions} name="checkboxOptions" 
      />);
    expect(toJson(searchinput)).toMatchSnapshot();
  })
})

