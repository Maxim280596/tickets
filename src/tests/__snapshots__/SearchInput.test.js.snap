// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SearchInput/> SearchInput have rendered correctly 1`] = `
<Field
  name="checkboxOptions"
  render={[Function]}
/>
`;
